tiller crashes on 'helm test' run when there's blank line in test pod yaml

When there's a blank line at the top of the test pod yaml template, tiller crashes on 'helm test' run.

Here's the problomatic (and rendered) yaml file:
```
(blank line)
---
apiVersion: v1
kind: Pod
metadata:
  name: "keystone-tempest-test"
  annotations:
    "helm.sh/hook": test-success
spec:
  containers:
...
```
If I run 'helm test' command, the following error is shown.
```
root@att-helm:~/workspace/openstack-helm# helm test keystone
Error: transport is closing
root@att-helm:~/workspace/openstack-helm# helm test keystone
Error: could not find a ready tiller pod
```
The tiller pod log says:
```
root@att-helm:~# kubectl logs tiller-deploy-3210613906-djz5c -n kube-system
Starting Tiller v2.2.3
GRPC listening on :44134
Probes listening on :44135
Storage driver is ConfigMap
2017/03/22 07:22:15 storage.go:133: Getting release history for 'keystone'
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xd44c8a]

goroutine 46 [running]:
panic(0x156c840, 0xc42000a030)
	/usr/local/go/src/runtime/panic.go:500 +0x1a1
k8s.io/helm/pkg/releasetesting.newTest(0xc42044b400, 0x0, 0x4110f8, 0x40, 0x161d1e0)
	/home/ubuntu/.go_workspace/src/k8s.io/helm/pkg/releasetesting/test_suite.go:174 +0x11a
k8s.io/helm/pkg/releasetesting.(*TestSuite).Run(0xc4203f9f40, 0xc420587b90, 0x0, 0x0)
	/home/ubuntu/.go_workspace/src/k8s.io/helm/pkg/releasetesting/test_suite.go:72 +0x9a
k8s.io/helm/pkg/tiller.(*ReleaseServer).RunReleaseTest(0xc42021ad00, 0xc42040b8a0, 0x1ff05c0, 0xc4204082e0, 0x0, 0x18)
	/home/ubuntu/.go_workspace/src/k8s.io/helm/pkg/tiller/release_server.go:1100 +0x244
k8s.io/helm/pkg/proto/hapi/services._ReleaseService_RunReleaseTest_Handler(0x16c8d20, 0xc42021ad00, 0x1fefcc0, 0xc420089440, 0xc4202becb8, 0x4110f8)
	/home/ubuntu/.go_workspace/src/k8s.io/helm/pkg/proto/hapi/services/tiller.pb.go:928 +0x117
k8s.io/helm/pkg/tiller.newStreamInterceptor.func1(0x16c8d20, 0xc42021ad00, 0x1fefcc0, 0xc420089440, 0xc42040b840, 0x17e0aa0, 0xc4202d5400, 0x4111d8)
	/home/ubuntu/.go_workspace/src/k8s.io/helm/pkg/tiller/server.go:63 +0x112
k8s.io/helm/vendor/google.golang.org/grpc.(*Server).processStreamingRPC(0xc4200d3080, 0x1ff0740, 0xc4200893b0, 0xc4200cb1d0, 0xc42043a720, 0x2035960, 0xc42029ff20, 0x0, 0x0)
	/home/ubuntu/.go_workspace/src/k8s.io/helm/vendor/google.golang.org/grpc/server.go:723 +0x2b6
k8s.io/helm/vendor/google.golang.org/grpc.(*Server).handleStream(0xc4200d3080, 0x1ff0740, 0xc4200893b0, 0xc4200cb1d0, 0xc42029ff20)
	/home/ubuntu/.go_workspace/src/k8s.io/helm/vendor/google.golang.org/grpc/server.go:800 +0xc36
k8s.io/helm/vendor/google.golang.org/grpc.(*Server).serveStreams.func1.1(0xc420408140, 0xc4200d3080, 0x1ff0740, 0xc4200893b0, 0xc4200cb1d0)
	/home/ubuntu/.go_workspace/src/k8s.io/helm/vendor/google.golang.org/grpc/server.go:449 +0xab
created by k8s.io/helm/vendor/google.golang.org/grpc.(*Server).serveStreams.func1
	/home/ubuntu/.go_workspace/src/k8s.io/helm/vendor/google.golang.org/grpc/server.go:450 +0xa3
```

This issue only takes place with 'helm test' command. 
If I remove the '"helm.sh/hook": test-success' from the above yaml file so that the pod can be launched by 'helm install' command, it just worked fine without any error.

Test Environment
- Helm version: 2.2.3
- OS: Ubuntu 16.04

--- a/pkg/releaseutil/manifest.go
+++ b/pkg/releaseutil/manifest.go
@@ -35,7 +35,6 @@ var sep = regexp.MustCompile("(?:^|\\s*\n)---\\s*")
 
 // SplitManifests takes a string of manifest and returns a map contains individual manifests
 func SplitManifests(bigfile string) map[string]string {
-	// This is not the best way of doing things, but it's how k8s itself does it.
 	// Basically, we're quickly splitting a stream of YAML documents into an
 	// array of YAML docs. In the current implementation, the file name is just
 	// a place holder, and doesn't have any further meaning.

--- a/pkg/releaseutil/manifest.go
+++ b/pkg/releaseutil/manifest.go
@@ -19,7 +19,6 @@ package releaseutil
 import (
 	"fmt"
 	"regexp"
-	"strings"
 )
 
 // SimpleHead defines what the structure of the head of a manifest file

--- a/pkg/releaseutil/manifest.go
+++ b/pkg/releaseutil/manifest.go
@@ -40,7 +40,9 @@ func SplitManifests(bigfile string) map[string]string {
 	sep := "\n---\n"
 	tpl := "manifest-%d"
 	res := map[string]string{}
-	tmp := strings.Split(bigfile, sep)
+	// Making sure yaml formatting doesn't matter when generating manifest from string.
+	bigFileTmp := strings.TrimSpace(bigfile)
+	tmp := strings.Split(bigFileTmp, sep)
 	for i, d := range tmp {
 		res[fmt.Sprintf(tpl, i)] = d
 	}

--- a/pkg/releaseutil/manifest.go
+++ b/pkg/releaseutil/manifest.go
@@ -32,19 +32,19 @@ type SimpleHead struct {
 	} `json:"metadata,omitempty"`
 }
 
+var sep = regexp.MustCompile("(?:^|\\s*\n)---\\s*")
+
 // SplitManifests takes a string of manifest and returns a map contains individual manifests
 func SplitManifests(bigfile string) map[string]string {
 	// This is not the best way of doing things, but it's how k8s itself does it.
 	// Basically, we're quickly splitting a stream of YAML documents into an
 	// array of YAML docs. In the current implementation, the file name is just
 	// a place holder, and doesn't have any further meaning.
-	sep := regexp.MustCompile("(?:^|\\s*\n)---\\s*")
 	tpl := "manifest-%d"
 	res := map[string]string{}
-	// Making sure YAML formatting doesn't matter when generating manifest from string.
-	bigFileTmp := strings.TrimSpace(bigfile)
-	tmp := sep.Split(bigFileTmp, -1)
-	for i, d := range tmp {
+	// Making sure that any extra whitespace in YAML stream doesn't interfere in splitting documents correctly.
+	docs := sep.Split(bigfile, -1)
+	for i, d := range docs {
 		res[fmt.Sprintf(tpl, i)] = d
 	}
 	return res

--- a/pkg/releaseutil/manifest.go
+++ b/pkg/releaseutil/manifest.go
@@ -18,6 +18,7 @@ package releaseutil
 
 import (
 	"fmt"
+	"regexp"
 	"strings"
 )
 
@@ -37,12 +38,12 @@ func SplitManifests(bigfile string) map[string]string {
 	// Basically, we're quickly splitting a stream of YAML documents into an
 	// array of YAML docs. In the current implementation, the file name is just
 	// a place holder, and doesn't have any further meaning.
-	sep := "\n---\n"
+	sep := regexp.MustCompile("(?:^|\\s*\n)---\\s*")
 	tpl := "manifest-%d"
 	res := map[string]string{}
-	// Making sure yaml formatting doesn't matter when generating manifest from string.
+	// Making sure YAML formatting doesn't matter when generating manifest from string.
 	bigFileTmp := strings.TrimSpace(bigfile)
-	tmp := strings.Split(bigFileTmp, sep)
+	tmp := sep.Split(bigFileTmp, -1)
 	for i, d := range tmp {
 		res[fmt.Sprintf(tpl, i)] = d
 	}

